runners/flink/src/main/java/org/apache/beam/runners/flink/translation/functions/FlinkMergingNonShuffleReduceFunction.java

```diff
@@ -83,7 +83,7 @@ public void reduce(
         new FlinkSideInputReader(sideInputs, getRuntimeContext());
 
     AbstractFlinkCombineRunner<K, InputT, AccumT, OutputT, W> reduceRunner;
-    if (windowingStrategy.getWindowFn().windowCoder().equals(IntervalWindow.getCoder())) {
+    if (windowingStrategy.needsMerge() && windowingStrategy.getWindowFn() instanceof Sessions) {
```

Comment:
I wonder if I was wrong actually and this is redundant. I would guess `FlinkMergingNonShuffleReduceFunction` will not be chosen unless `needsMerge` is already known to be true. Sorry I missed the context for this change. I think it is fine, but maybe not needed after all.

Reply:
yeah, looks like FlinkMergingNonShuffleReduceFunction is only used when the status is needsMerge.https://github.com/apache/beam/blob/master/runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkBatchTransformTranslators.java#L593.